cmake/kokkos_arch.cmake

```diff
@@ -1163,6 +1163,26 @@ if(KOKKOS_ARCH_HOPPER90)
   set(KOKKOS_ARCH_HOPPER ON)
 endif()
 
+function(CHECK_AMD_APU ARCH)
+  set(BINARY_TEST_DIR ${CMAKE_CURRENT_BINARY_DIR}/cmake/compile_tests/AmdApuWorkdir)
+  file(REMOVE_RECURSE ${BINARY_TEST_DIR})
+  file(MAKE_DIRECTORY ${BINARY_TEST_DIR})
+
+  try_run(RESULT COMPILE_RESULT ${BINARY_TEST_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/compile_tests/amd_apu.cc
+          RUN_OUTPUT_VARIABLE AMD_APU
+  )
+
+  if(NOT COMPILE_RESULT OR NOT RESULT EQUAL 0)
+    message(SEND_ERROR "Autodetection of AMD APU failed."
+                       "Please manually specify one AMD GPU architecture via -DKokkos_ARCH_{..}=ON'."
+    )
+  endif()
+
+  if(AMD_APU)
+    set(${ARCH} AMD_GFX942_APU PARENT_SCOPE)
```

Comment:
Maybe we should check here also that the little agent reporter return gfx942? Not sure if there are other APUs, now or in the future?

Reply:
ah never mind we only do this if its already gfx942 sounds good, we can later expand if and when needed.

Reply:
I didn't want to make the code more general/complicated not knowing if this will be a one-off.

Reply:
As I said its fine, we are already inside a gfx942 clause and we can generalize if and when necessary.